### PR TITLE
Better pie chart legend

### PR DESCRIPTION
**Pull request by rdwebdesign**
**By submitting this pull request, I confirm the following:** 

- [x] I have read and understood the [contributors guide](https://github.com/pi-hole/AdminLTE/blob/master/CONTRIBUTING.md), as well as this entire template.
- [x] I have made only one major change in my proposed changes.
- [x] I have commented my proposed changes within the code.
- [x] I have tested my proposed changes.
- [x] I am willing to help maintain this change if there are issues with it later.
- [x] I give this submission freely and claim no ownership.
- [x] It is compatible with the [EUPL 1.2 license](https://opensource.org/licenses/EUPL-1.1)
- [x] I have squashed any insignificant commits. ([`git rebase`](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html))
- [x] I have Signed Off all commits. ([`git commit --signoff`](https://git-scm.com/docs/git-commit#Documentation/git-commit.txt---signoff))

---

**What does this PR aim to accomplish?:**

This PR improves the pie chart legend.

**How does this PR accomplish the above?:**
dashboard_piechart_legend

Improvements:

- Cleaner legend text (hiding the block color);
- better text wrapping for long series names;
- hover effects and title parameters to the links;

**What documentation changes (if any) are needed to support this PR?:**

none

### CODE REVIEW

**Comment by DL6ER** [comment]
ezgif-2-01378af22d

**Review by yubiuser — DISMISSED** [review]

File: scripts/pi-hole/js/index.js [hunk 1]
```diff
@@ -319,22 +319,69 @@ function updateQueryTypesPie() {
...
     $("#query-types-legend").html(queryTypePieChart.generateLegend());
-    $("#query-types-legend > ul > li").prepend(createEyeConElement());
     $("#query-types-legend > ul > li").click(function (e) {
       if (isEyeCon(e.target)) {
         return false;
       }
 
       window.location.href = "queries.php?querytype=" + querytypeids[$(this).index()];
     });
+    $("#query-types-legend .eyeConWrapper").click(function (e) {
```
Comment (yubiuser) [inline scripts/pi-hole/js/index.js]:
```suggestion
    $("#query-types-legend .checkboxWrapper").click(function (e) {
```

Comment (yubiuser) [reply scripts/pi-hole/js/index.js]: All the eye needs to be replaced.

**Review by yubiuser — CHANGES_REQUESTED** [review]

File: scripts/pi-hole/js/index.js (hunk 1 above)
Comment (rdwebdesign) [reply scripts/pi-hole/js/index.js]: I will use a different name. 
`checkboxWrapper` is not the right name for this legend box.

`colorboxWrapper` is a better name.

**Comment by DL6ER** [comment]
Thank you very much. The screenshot in the PR description is outdated, new screenshots:

Screenshot from 2022-01-16 12-24-09
Screenshot from 2022-01-16 12-25-44

With some disabled types:
Screenshot from 2022-01-16 12-24-45
Screenshot from 2022-01-16 12-25-57

**Comment by DL6ER** [comment]
One things I noticed is that the updates every 60 seconds undo the `hidePieSlice()` action, i.e., if you deselected some types, the periodic update resets this to showing all.

**Comment by rdwebdesign** [comment]
@DL6ER 

Yes. `setTimeout()` is loading new data and resetting the whole graphic (including the legend). This was happening before, but was hidden until #1996.

I we can solve this in a new PR.

**Review by DL6ER — APPROVED** [review]

**Review by yubiuser — APPROVED** [review]

**Pull request merged**